### PR TITLE
Hw_4.decorator

### PR DESCRIPTION
**Pull request by ninaprus**
@divide_by_100
@type_arg
@cache_decorator

### CODE REVIEW

File: hw_decorator.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+# Если это int, тогда выполнить функцию и вывести результат, если это str(),
+# тогда зарейзить ошибку ValueError (raise ValueError(“string type is not supported”))
+
+def type_arg(func):
+    @functools.wraps(func)
+    def wrapper_type_arg(args):
+        if type(args)==int:
+            value = func(args)
+            return value
+        elif type(args)==str:
```
Comment (katevoskresenska) [inline hw_decorator.py]: It is better to use isinstance(args, int), isinstance(args, str)

File: hw_decorator.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+def my_count_func(number):
+    res = sum([i for i in range(number)])
+    return res
+
+my_count_func(10)
+my_count_func(6)
+my_count_func(6)
+my_count_func(10)
+my_count_func(8)
+my_count_func(10)
```
Comment (katevoskresenska) [inline hw_decorator.py:86]: It would be nice to add some documentation strings to all the functions.

Comment (ninaprus) [reply hw_decorator.py:86]: Thanks, I did

File: hw_decorator.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+def type_arg(func):
+    """Decorator checks the type argument that is passed to your function."""
+    @functools.wraps(func)
+    def wrapper_type_arg(args):
+        if isinstance(args, int):
+            value = func(args)
+            return value
+        elif isinstance(args, str):
+            raise ValueError('String type is not supported')
+        else:
```
Comment (mmykhailenko) [inline hw_decorator.py]: could remove else block

**Review by mmykhailenko — APPROVED** [review]

**Pull request merged**